gizmo.py

```diff
@@ -0,0 +1,90 @@
+## open python in terminal, import gizmo, call function
+## https://github.com/MertenNefs/gizmo/tree/testbranch
+
+## exercise 1
+import gizmo
+
+## exercise 2
+def hello(name,country='Finland'): 
+    print('Hello '+ name + ', how are things in ' + country + '?')
+    
+## exercise 3
+def spell(name):
+    spelled = ''
+    for l in name:
+        spelled += l + '.' # spelled + l + '.'
+    print(spelled[:-1])
+    
+## exercise 4 
+### https://realpython.com/python-f-strings/
+def relative_path(subject_identifiers):
+    paths = []
+    for i in subject_identifiers:
+        paths.append(f"./subjects/mock_recording_{i}.rec")
+    return paths
+
+## exercise 5/6/7 add attribute and method to class
+class Gizmo:
+    def __init__(self, name):
+        self.name = name
+    def speak(self):
+        print(self.name)
+        
+## exercise 8/9/10 numpy
+import numpy as np
+def multiplication_table(zero_out_multiples = None):
+    """
+    Summary: Generates a table of multiples
+
+    Extended Summary: As an outer product of the array x,
+    eliminating multiples of the argument.
+    
+    Parameters
+    ----------
+    zero_out_multiples : int
+        argument of integer value to zero out.
+   
+    Returns
+    -------
+    y : array
+        outer product array of integers resulting from (x,x).
+ 
+    """
+    x = np.array([1,2,3,4,5,6,7,8,9,10,11,12])
+    y = np.outer(x,x)
+    if zero_out_multiples != None:
```

Comment:
While this works, the recommended "pythonic" way to check against `None` is with the `is` keyword. This is because `==` compares values (converting int -> float and such as necessary), which is what you want with numbers and strings, and 'is' compares object identities. Since `None` is a special singleton object, all `None` objects are one and the same object. So, technically, you should use `is` to compare them. The nitpickiest of nitpicks!

```python
if zero_out_multiples is not None:
```